docs/notifier-catalog.md

```diff
@@ -1746,10 +1738,151 @@ Requires:
 
 ### `ntfy`
 
-Support for [ntfy] is provided through Apprise, see [apprise_single](#apprise_single)
-and [apprise_multi](#apprise_multi).
+> [ntfy] (pronounce: _notify_) is a simple HTTP-based [pub-sub] notification service.
+> It allows you to send notifications to your phone or desktop via scripts from
+> any computer, entirely without signup, cost or setup.
+> [ntfy is also open source](https://github.com/binwiederhier/ntfy), if you want to
+> run an instance on your own premises.
+
+ntfy uses topics to address communication channels. This topic is part of the
+HTTP API URL.
+
+To use the hosted variant on `ntfy.sh`, just provide an URL including the topic.
+```ini
+[config:ntfy]
+targets  = {
+    'test': 'https://ntfy.sh/testdrive',
+    }
+```
+
+When running your own instance, you would use a custom URL here.
+```ini
+[config:ntfy]
+targets  = {
+    'test': 'http://username:password@localhost:5555/testdrive',
+    }
+```
+
+In order to specify more options, please wrap your ntfy URL into a dictionary
+under the `url` key. This way, additional options can be added.
+```ini
+[config:ntfy]
+targets  = {
+    'test': {
+        'url': 'https://ntfy.sh/testdrive',
+        },
+    }
+```
+
+:::{important}
+[ntfy publishing options] outlines different ways to marshal data to the ntfy
+HTTP API. mqttwarn is using HTTP headers for serializing values, because the
+HTTP body will already be used for the attachment file. Because of this, you
+are not able to use UTF-8 characters within your message text, they will be
+replaced by placeholder characters like `?`.
+:::
+
+{#ntfy-remote-attachments}
+#### Remote attachments
+In order to submit notifications with an attachment file at a remote location,
+use the `attach` field. Optionally, the `filename` field can be used to assign
+a different name to the file.
+```ini
+[config:ntfy]
+targets  = {
+    'test': {
+        'url': 'https://ntfy.sh/testdrive',
+        'attach': 'https://unsplash.com/photos/spdQ1dVuIHw/download?w=320',
+        'filename': 'goat.jpg',
+        },
+    }
+```
+
+{#ntfy-local-attachments}
+#### Local attachments
+By using the `attachment` option, you can add an attachment to your message, local
+to the machine mqttwarn is running on. The file will be uploaded when submitting
+the notification, and ntfy will serve it for clients so that you don't have to. In
+order to address the file, you can provide a path template, where the transformation
+data will also get interpolated into.
+```ini
+[config:ntfy]
+targets  = {
+    'test': {
+        'url': 'https://ntfy.sh/testdrive',
+        'attachment': '/tmp/ntfy-attachment-{slot}-{label}.png',
```

Comment:
I've introduced the `attachment` option/field to signal to mqttwarn that it should [attach a local file](https://docs.ntfy.sh/publish/#attach-local-file) to the ntfy notification, in order to support @sevmonster's use case.

However, I do not like the naming yet, as it's easy to confuse with the `attach` field. We should choose another name. I am leaning towards just using `file` instead. Do you have any other suggestions?

Reply:
Renamed with 986fcb6a77.